Rubberduck.Core/UI/DockableToolwindowPresenter.cs

```diff
@@ -108,27 +108,34 @@ private void EnsureMinimumWindowSize(IWindow window)
         public virtual void Show() => _window.IsVisible = true;
         public virtual void Hide() => _window.IsVisible = false;
 
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
 
         private bool _isDisposed;
-        public void Dispose()
+        protected virtual void Dispose(bool disposing)
         {
-            if (_isDisposed)
+            if (_isDisposed || !disposing)
             {
                 return;
             }
 
             Logger.Trace($"Disposing DockableWindowPresenter of type {this.GetType()}.");
 
             _window.Dispose();
-
+           
             _isDisposed = true;
         }
 
 #if DEBUG
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1063")] // only logging here.
         ~DockableToolwindowPresenter()
         {
             // destructor for tracking purposes only - do not suppress unless 
```

Comment:
FWIW the comment seem to contradict the current behavior. We are now using the finalizer to ensure disposal in addition to logging (e.g. tracking)

Reply:
Not sure what I was thinking there... fixed.